src/coreclr/clrfeatures.cmake

```
@@ -38,4 +38,20 @@ endif()
 
 if (CLR_CMAKE_TARGET_WIN32)
   set(FEATURE_TYPEEQUIVALENCE 1)
-endif(CLR_CMAKE_TARGET_WIN32)
\ No newline at end of file
+endif(CLR_CMAKE_TARGET_WIN32)
+
+# Enable cached interface dispatch so that we can test/debug it more easily on non-embedded scenarios (set DOTNET_UseCachedInterfaceDispatch=1)
+# Only enable in chk/debug builds as this support isn't intended for retail use elsewhere
+if (((CMAKE_BUILD_TYPE STREQUAL "Debug") OR (CMAKE_BUILD_TYPE STREQUAL "Checked")) AND (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64))
+  if (CLR_CMAKE_HOST_UNIX AND CLR_CMAKE_HOST_ARCH_AMD64)
+    # Allow 16 byte compare-exchange (cmpxchg16b)
+    add_compile_options(-mcx16)
+  endif()
+  set(FEATURE_CORECLR_CACHED_INTERFACE_DISPATCH 1)
```

nit: maybe
```suggestion
  set(FEATURE_CACHED_INTERFACE_DISPATCH 1)
```
or `FEATURE_CLR_CACHED_INTERFACE_DISPATCH` if it is necessary to qualify it.

---

@am11 I can see we already have FEATURE_CACHED_INTERFACE_DISPATCH, so I guess this is to differentiate between the new feature in coreclr (thus the name) and the nativeaot one - the preexisting one.

---

Correct. We can't set the same feature flag for NativeAOT and for CoreCLR since we need the AOT one to be on for Linux X64 and off for CoreCLR Linux X64, and having a distinct name is actually kinda important. And I try very hard to avoid using CLR instead of CORECLR, since CLR internally to Microsoft has a specific meaning of referring to the runtime for .NET Framework.